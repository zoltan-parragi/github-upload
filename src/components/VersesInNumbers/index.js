import React from "react";

const VersesInNumbers = ({ versesTotal, minVerseNumberOnPage, maxVerseNumberOnPage }) => 
	<div className = "versesinfo-numbers">
	    <span className="versesinfo-numbers-total">The number of verses: {versesTotal}</span>
	    {   (versesTotal === 0)
			? <span className="versesinfo-numbers-page">page: {1}</span>
			: <span className="versesinfo-numbers-page">page: {(minVerseNumberOnPage)/10 + 1}</span>
		}
		{   (versesTotal === 0)
			? <span className="versesinfo-numbers-verses">verses: {0} </span>
			: (versesTotal > maxVerseNumberOnPage) 
			? <span className="versesinfo-numbers-verses">verses: {minVerseNumberOnPage + 1} - {maxVerseNumberOnPage}</span> 
			: <span className="versesinfo-numbers-verses">verses: {minVerseNumberOnPage + 1} - {versesTotal}</span>
		}
	</div>

export default VersesInNumbers;