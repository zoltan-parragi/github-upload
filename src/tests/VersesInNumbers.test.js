import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import VersesInNumbers from '../components/VersesInNumbers';

describe('VersesInNumbers', () => {
	it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(<VersesInNumbers />, div);
	ReactDOM.unmountComponentAtNode(div);
	});
	test('has valid snapshot', () => {
		const component = renderer.create(<VersesInNumbers />);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});