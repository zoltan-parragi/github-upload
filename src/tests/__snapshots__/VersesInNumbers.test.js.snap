// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VersesInNumbers has valid snapshot 1`] = `
<div
  className="versesinfo-numbers"
>
  <span
    className="versesinfo-numbers-total"
  >
    The number of verses: 
  </span>
  <span
    className="versesinfo-numbers-page"
  >
    page: 
    NaN
  </span>
  <span
    className="versesinfo-numbers-verses"
  >
    verses: 
    NaN
     - 
  </span>
</div>
`;
